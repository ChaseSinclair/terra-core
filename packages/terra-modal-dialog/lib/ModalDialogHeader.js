'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _ModalDialogHeaderDismiss = require('./ModalDialogHeaderDismiss');

var _ModalDialogHeaderDismiss2 = _interopRequireDefault(_ModalDialogHeaderDismiss);

require('./ModalDialogHeader.scss');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var propTypes = {
  children: _propTypes2.default.node,
  hasDismiss: _propTypes2.default.node,
  onClick: _propTypes2.default.func
};

var defaultProps = {
  children: null
};

var ModalDialogHeader = function ModalDialogHeader(_ref) {
  var children = _ref.children,
      onClick = _ref.onClick,
      customProps = _objectWithoutProperties(_ref, ['children', 'onClick']);

  var dismissButton = onClick ? _react2.default.createElement(_ModalDialogHeaderDismiss2.default, { onClick: onClick }) : null;

  return _react2.default.createElement(
    'div',
    { className: 'terra-ModalDialog-header' },
    _react2.default.createElement(
      'div',
      _extends({ className: 'terra-ModalDialog-header-title' }, customProps),
      children
    ),
    dismissButton
  );
};

ModalDialogHeader.propTypes = propTypes;
ModalDialogHeader.defaultProps = defaultProps;

exports.default = ModalDialogHeader;