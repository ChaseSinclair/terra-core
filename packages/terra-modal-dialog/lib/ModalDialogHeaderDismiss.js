'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _IconClose = require('terra-icon/lib/icon/IconClose');

var _IconClose2 = _interopRequireDefault(_IconClose);

var _terraButton = require('terra-button');

var _terraButton2 = _interopRequireDefault(_terraButton);

require('./ModalDialogDismiss.scss');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var propTypes = {
  children: _propTypes2.default.node,
  onClick: _propTypes2.default.func
};

var defaultProps = {
  children: null
};

var ModalDialogDismiss = function ModalDialogDismiss(_ref) {
  var onClick = _ref.onClick,
      customProps = _objectWithoutProperties(_ref, ['onClick']);

  return _react2.default.createElement(
    _terraButton2.default,
    _extends({ variant: 'link', className: 'terra-ModalDialog-header-dismiss', onClick: onClick }, customProps),
    _react2.default.createElement(_IconClose2.default, null)
  );
};

ModalDialogDismiss.propTypes = propTypes;
ModalDialogDismiss.defaultProps = defaultProps;

exports.default = ModalDialogDismiss;