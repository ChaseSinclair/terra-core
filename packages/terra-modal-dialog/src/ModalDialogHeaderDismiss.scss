@import '~terra-mixins';

.terra-ModalDialog-header-dismiss {
  @include terra-padding-horizontal(0, 0);
  border: 0;
  color: $terra-dark;
  cursor: pointer;
  height: 2em;
  margin-bottom: 0.3571rem;
  margin-left: 0.7143rem;
  margin-right: 0.7143rem;
  margin-top: 0.3571rem;
  width: 2em;
  display: table-cell;
}
