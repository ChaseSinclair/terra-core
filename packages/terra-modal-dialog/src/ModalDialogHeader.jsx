import React from 'react';
import PropTypes from 'prop-types';
import ModalDialogHeaderDismiss from './ModalDialogHeaderDismiss';
import './ModalDialogHeader.scss';

const propTypes = {
  children: PropTypes.node,
  hasDismiss: PropTypes.node,
  onClick: PropTypes.func,
};

const defaultProps = {
  children: null,
};

const ModalDialogHeader = ({ children, onClick, ...customProps }) => {
  const dismissButton = onClick ? <ModalDialogHeaderDismiss onClick={onClick} /> : null;

  return (<div className={'terra-ModalDialog-header'}>
    <div className={'terra-ModalDialog-header-title'} {...customProps} >
      {children}
    </div>
    {dismissButton}
  </div>);
};

ModalDialogHeader.propTypes = propTypes;
ModalDialogHeader.defaultProps = defaultProps;

export default ModalDialogHeader;
