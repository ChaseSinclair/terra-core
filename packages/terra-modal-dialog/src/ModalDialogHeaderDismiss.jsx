import React from 'react';
import PropTypes from 'prop-types';
import IconClose from 'terra-icon/lib/icon/IconClose';
import Button from 'terra-button';
import './ModalDialogHeaderDismiss.scss';

const propTypes = {
  children: PropTypes.node,
  onClick: PropTypes.func,
};

const defaultProps = {
  children: null,
};

const ModalDialogDismiss = ({ onClick, ...customProps }) => (
  <Button variant="link" className="terra-ModalDialog-header-dismiss" onClick={onClick} {...customProps} >
    <IconClose />
  </Button>);

ModalDialogDismiss.propTypes = propTypes;
ModalDialogDismiss.defaultProps = defaultProps;

export default ModalDialogDismiss;
