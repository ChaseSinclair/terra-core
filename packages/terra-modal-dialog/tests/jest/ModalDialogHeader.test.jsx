import React from 'react';
import ModalDialogHeader from '../../src/ModalDialogHeader';
import ModalDialogHeaderDismiss from '../../src/ModalDialogHeaderDismiss';

describe('ModalDialogHeader with dismiss button', () => {
  const modalDialogHeader = <ModalDialogHeader onClick={() => console.log('clicked')}>This is content</ModalDialogHeader>;

  it('should shallow a modal dialog header', () => {
    const modal = shallow(modalDialogHeader);
    expect(modal).toMatchSnapshot();
  });

  it('should mount a modal dialog header', () => {
    const modal = mount(modalDialogHeader);
    expect(modal).toMatchSnapshot();
  });

  it('should render a modal dialog header', () => {
    const modal = render(modalDialogHeader);
    expect(modal).toMatchSnapshot();
  });

  it('should have class terra-ModalDialog-header', () => {
    const modal = shallow(modalDialogHeader);
    expect(modal.prop('className')).toEqual('terra-ModalDialog-header');
  });

  it('should have a dismiss button', () => {
    const modal = shallow(modalDialogHeader);
    expect(modal.find(ModalDialogHeaderDismiss).exists()).toEqual(true);
  });
});

describe('ModalDialogHeader without dismiss button', () => {
  const modalDialogHeader = <ModalDialogHeader>This is content</ModalDialogHeader>;

  it('should shallow a modal dialog header', () => {
    const modal = shallow(modalDialogHeader);
    expect(modal).toMatchSnapshot();
  });

  it('should mount a modal dialog header', () => {
    const modal = mount(modalDialogHeader);
    expect(modal).toMatchSnapshot();
  });

  it('should render a modal dialog header', () => {
    const modal = render(modalDialogHeader);
    expect(modal).toMatchSnapshot();
  });

  it('should have class terra-ModalDialog-header', () => {
    const wrapper = shallow(modalDialogHeader);
    expect(wrapper.prop('className')).toEqual('terra-ModalDialog-header');
  });

  it('should not have a dismiss button', () => {
    const modal = shallow(modalDialogHeader);
    expect(modal.find(ModalDialogHeaderDismiss).exists()).toEqual(false);
  });
});
