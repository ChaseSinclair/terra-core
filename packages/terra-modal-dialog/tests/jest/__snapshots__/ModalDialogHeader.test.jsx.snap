exports[`ModalDialogHeader with dismiss button should mount a modal dialog header 1`] = `
<ModalDialogHeader
  onClick={[Function]}>
  <div
    className="terra-ModalDialog-header">
    <div
      className="terra-ModalDialog-header-title">
      This is content
    </div>
    <ModalDialogDismiss
      onClick={[Function]}>
      <Button
        className="terra-ModalDialog-header-dismiss"
        isBlock={false}
        isCompact={false}
        isDisabled={false}
        isReversed={false}
        onClick={[Function]}
        variant="link">
        <button
          aria-disabled={false}
          className="terra-Button terra-Button--link terra-ModalDialog-header-dismiss"
          disabled={false}
          onClick={[Function]}>
          <IconClose
            data-name="Layer 1"
            viewBox="0 0 48 48"
            xmlns="http://www.w3.org/2000/svg">
            <IconBase
              ariaLabel={null}
              data-name="Layer 1"
              focusable={false}
              height="1em"
              isBidi={false}
              isSpin={false}
              viewBox="0 0 48 48"
              width="1em"
              xmlns="http://www.w3.org/2000/svg">
              <svg
                aria-hidden="true"
                className="terra-Icon"
                data-name="Layer 1"
                focusable={false}
                height="1em"
                viewBox="0 0 48 48"
                width="1em"
                xmlns="http://www.w3.org/2000/svg">
                <path
                  d="M28.2 24L42.9 9.1 40.8 7l-1.7-1.6-.4-.5L24 19.7 9.4 4.9 7.2 7 5.6 8.6l-.5.5L19.8 24 5.1 38.9 7.2 41l1.7 1.6.5.5L24 28.3l14.7 14.8.4-.5 1.7-1.6 2.1-2.1L28.2 24z" />
              </svg>
            </IconBase>
          </IconClose>
        </button>
      </Button>
    </ModalDialogDismiss>
  </div>
</ModalDialogHeader>
`;

exports[`ModalDialogHeader with dismiss button should render a modal dialog header 1`] = `
<div
  class="terra-ModalDialog-header">
  <div
    class="terra-ModalDialog-header-title">
    This is content
  </div>
  <button
    aria-disabled="false"
    class="terra-Button terra-Button--link terra-ModalDialog-header-dismiss">
    <svg
      aria-hidden="true"
      class="terra-Icon"
      data-name="Layer 1"
      focusable="false"
      height="1em"
      viewbox="0 0 48 48"
      width="1em"
      xmlns="http://www.w3.org/2000/svg">
      <path
        d="M28.2 24L42.9 9.1 40.8 7l-1.7-1.6-.4-.5L24 19.7 9.4 4.9 7.2 7 5.6 8.6l-.5.5L19.8 24 5.1 38.9 7.2 41l1.7 1.6.5.5L24 28.3l14.7 14.8.4-.5 1.7-1.6 2.1-2.1L28.2 24z" />
    </svg>
  </button>
</div>
`;

exports[`ModalDialogHeader with dismiss button should shallow a modal dialog header 1`] = `
<div
  className="terra-ModalDialog-header">
  <div
    className="terra-ModalDialog-header-title">
    This is content
  </div>
  <ModalDialogDismiss
    onClick={[Function]} />
</div>
`;

exports[`ModalDialogHeader without dismiss button should mount a modal dialog header 1`] = `
<ModalDialogHeader>
  <div
    className="terra-ModalDialog-header">
    <div
      className="terra-ModalDialog-header-title">
      This is content
    </div>
  </div>
</ModalDialogHeader>
`;

exports[`ModalDialogHeader without dismiss button should render a modal dialog header 1`] = `
<div
  class="terra-ModalDialog-header">
  <div
    class="terra-ModalDialog-header-title">
    This is content
  </div>
</div>
`;

exports[`ModalDialogHeader without dismiss button should shallow a modal dialog header 1`] = `
<div
  className="terra-ModalDialog-header">
  <div
    className="terra-ModalDialog-header-title">
    This is content
  </div>
</div>
`;
